src/common/Tabs/Tabs.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { getNestedObject } from '@utils/utils';
+import Tab from './Tab';
+/** Styled components */
+import { TabsWrapper, TabList, TabContent } from './tabs.styled';
+
+const Tabs = ({ children }) => {
+  const [tabs, setTabs] = useState([]);
+  const [activeTab, setActiveTab] = useState('');
+
+  useEffect(() => {
+    if (Array.isArray(children) && children.length) {
+      setActiveTab(getNestedObject(children[0], ['props', 'label']));
+      setTabs(children);
+    } else if (typeof children === 'object') {
+      setActiveTab(getNestedObject(children, ['props', 'label']));
+      setTabs([children]);
+    }
+  }, [children, setTabs, setActiveTab, getNestedObject]);
```

Comment:
Why so many deps? I think the only one needed is `children`. We should check this.

Reply:
That's a good question. I seem to remember talking to you about these things before. In the useEffect, you must pass all the dependencies in the array. Why? [](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies)

I know because I asked myself the same question when ESLint warned me.

Reply:
Yes, but the setters are not part of the dependencies. The only value that you are using for real is `children`. The ESLint rule is not extremely intelligent to determine what may or may not change.

Reply:
I agree that eslint is not smart enough to determine which values change and which do not, but the setters are part of the dependencies, because they are used inside. 

We can avoid passing it as a dependency and disable the warning because, in this case, we know that they never changes. In my opinion they dont disturb but I can delete them np.

On the other side, I cant reproduce the visual bug.

![image](https://user-images.githubusercontent.com/15815193/75814706-80754980-5d92-11ea-8551-75cfaa2e3099.png)

Reply:
> but the setters are part of the dependencies, because they are used inside

I strongly disagree.

Let me put a quick example and see it for yourself.

```javascript
useEffect(() => {
  setCount(count => count + step );
}, [step]);
```
In the code above, the _useEffect_ is triggered every time that `step` changes. Here we are using a setter, `setCount`, but that doesn't make it a dependency, `step` does because we are using its value. Then, putting `setCount` as a dep is wrong.

In any case, both of us have seen a lot of React code out there and I'm pretty sure you know putting setters as dependencies is quite suspicious and... smelly. It just doesn't make sense. And TBH, I never saw it anywhere, and I'm pretty sure you didn't either.

Fuck the ESLint rule.

Reply:
In my opinion they are dependecies. Your example demonstrates that they are not necesary.

However, I delete them and checked the styles. The two files I touched (Queue.scss and sidebar.styled.js)  don't change the interface a priori.

Reply:
> In my opinion they are dependencies. Your example demonstrates that they are not necesary.

They are not necessary, ergo they are not dependencies.

Btw, I saw we have the `eslint-plugin-react-hooks` in the `package.json`, but we don't have one of the most important rules activated in the project: `exhaustive-deps`. If we want to rely on ESLint about Hooks dependencies we should put the following in the eslint config file:

```javascript
{
  "plugins": ["react-hooks"],
  // ...
  "rules": {
    "react-hooks/rules-of-hooks": 'error',
    "react-hooks/exhaustive-deps": 'warn' // <--- THIS IS THE NEW RULE
  }
}
```

I've done it in my local machine and ESLint found a few missing _useEffect_ dependencies that maybe we have to take a look.

FYI, in the _useEffect_ inside `Tabs`, the only suggested dep was `children`.